packages/@aws-cdk/aws-cognito-identitypool/test/identitypool.test.ts

```diff
@@ -0,0 +1,539 @@
+import {
+  Template,
+} from '@aws-cdk/assertions';
+import {
+  UserPool,
+  UserPoolIdentityProvider,
+} from '@aws-cdk/aws-cognito';
+import {
+  Role,
+  ServicePrincipal,
+  ArnPrincipal,
+  AnyPrincipal,
+  OpenIdConnectProvider,
+  SamlProvider,
+  SamlMetadataDocument,
+  PolicyStatement,
+  Effect,
+  PolicyDocument,
+} from '@aws-cdk/aws-iam';
+import {
+  Stack,
+} from '@aws-cdk/core';
+import {
+  IdentityPool,
+  IdentityPoolProviderUrl,
+} from '../lib/identitypool';
+import {
+  RoleMappingMatchType,
+} from '../lib/identitypool-role-attachment';
+import { UserPoolAuthenticationProvider } from '../lib/identitypool-user-pool-authentication-provider';
+
+describe('identity pool', () => {
+  test('minimal setup', () => {
+    const stack = new Stack();
+    new IdentityPool(stack, 'TestIdentityPoolMinimal');
+    const temp = Template.fromStack(stack);
+
+    temp.hasResourceProperties('AWS::Cognito::IdentityPool', {
+      AllowUnauthenticatedIdentities: false,
+    });
+
+    temp.resourceCountIs('AWS::IAM::Role', 2);
+    temp.resourceCountIs('AWS::IAM::Policy', 0);
+    temp.hasResourceProperties('AWS::IAM::Role', {
```

Comment:
Should we add another one for the unauth role?

Reply:
Done